tests/legacy-cli/e2e/utils/wsl.ts

```diff
@@ -0,0 +1,107 @@
+import assert from 'node:assert';
+import fs from 'node:fs';
+import path from 'node:path';
+import child_process from 'node:child_process';
+import { getGlobalVariable } from './env';
+
+interface WindowsWslTestMode {
+  cmdPath: string;
+  gitBinaryDirForWindows: string;
+  nodeBinaryForWindowsPath: string;
+  nodeBinaryForWindowsInsideWslPath: string;
+  npmBinaryForWindowsPath: string;
+  wslRootPath: string;
+  wslUncBase: string;
+}
+
+export const windowsTmpDir = process.env['NG_E2E_RUNNER_WINDOWS_TMP_DIR'];
+
+export function shouldInteropWslPath(p: string): boolean {
+  return p.startsWith('/') && fs.existsSync(p);
+}
+
+export function interopWslPathForOutsideIfNecessary(p: string): string {
+  const windowsMode = isWindowsTestMode();
+  assert(windowsMode !== null);
+
+  if (shouldInteropWslPath(p)) {
+    return (
+      child_process
+        .execSync(`wslpath -w ${p}`, { encoding: 'utf8' })
+        .trim()
+        // Note: We expect a fake symlinked WSL root directory in our host's
+        // main drive. This is necessary because `npm publish` does not work
+        // with UNC paths, like `\\wsl.localhost\Debian\<..>`.
+        .replace(windowsMode.wslUncBase, windowsMode.wslRootPath)
+    );
+  }
+  return p;
+}
+
+export function isWindowsTestMode(): WindowsWslTestMode | null {
```

Comment:
Should this be memoized in some way? Not that its extremely costly, but we run this check a bunch and don't need to constantly do assertions.

Might not be worth the effort, but I figured I would bring it up.

Reply:
For sure. I also wanted to do this, but didn't get to it yet.